File: lc1/two_sum.py
class Solution(object):
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        """
        Given an array of integers nums and an integer target,
        return indices of the two numbers such that they add up to target.
                :type nums: List[int]
                :type target: int
                :rtype: List[int]
        """
        sum_dict = {}
        for i, num in enumerate(nums):
            sum = target - num
            if sum_dict.get(sum, -1) >= 0:
                return [sum_dict[sum], i]
            sum_dict[num] = i
        return []

File: lc1/two_sum.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> target_map;
    vector<int> result;

    for (int i = 0; i < nums.size(); i++) {
      int pair = target - nums[i];
      target_map[pair] = i;
    }

    for (int i = 0; i < nums.size(); i++) {
      if (target_map.count(nums[i]) && target_map[nums[i]] != i) {
        result = {i, target_map[nums[i]]};
        break;
      }
    }
    return result;
  }
};
